frontend/packages/dev-console/src/components/pipelines/PipelineResources.tsx

```diff
@@ -35,39 +42,52 @@ const PipelineResources: React.FC<FormikProps<FormikValues>> = ({
           addLabel="Add Pipeline Resources"
           headers={['Name', 'Resource Type']}
           emptyValues={{ name: '', type: '' }}
+          isReadOnly={!pipelineResourceAccess}
         >
-          <InputField name="name" type={TextInputTypes.text} placeholder="Name" />
-          <DropdownField name="type" items={resourceTypes} fullWidth />
+          <InputField
+            name="name"
+            type={TextInputTypes.text}
+            placeholder="Name"
+            isReadOnly={!pipelineResourceAccess}
+          />
+          <DropdownField
+            name="type"
+            items={resourceTypes}
+            fullWidth
+            disabled={!pipelineResourceAccess}
```

Comment:
For consistency we probably want to add a `isReadOnly` field to `DropdownField`.
In html the attribute is `readonly`. I prefer to also be consistent but I can be swayed here.
cc @rohitkrai03